docs/topics/tasks.txt

```diff
@@ -0,0 +1,337 @@
+functions.
+
+Depending on the backend configured, it may be necessary to gracefully degrade
+functionality, or use a system check to prevent the application from starting at all.
+
+To facilitate this, certain features can be checked on a backend:
+
+* :attr:`supports_defer <django.tasks.backends.base.BaseTaskBackend.supports_defer>`:
+  Can tasks be executed after a specific time using ``run_after``?
+
+* :attr:`supports_async_task <django.tasks.backends.base.BaseTaskBackend.supports_async_task>`:
+  Can ``async`` functions (coroutines) be used as task functions?
+
+* :attr:`supports_get_result <django.tasks.backends.base.BaseTaskBackend.supports_get_result>`:
+  Can a task's results be retrieved from another thread or process?
+
+Defining tasks
+==============
+
+Tasks are defined using the :meth:`django.tasks.task` decorator on a
+module-level function::
+
+    from django.tasks import task
+
+
+    @task()
+    def calculate_meaning_of_life(answer=42):
+        return answer
+
+Returned in a :class:`django.tasks.Task` instance.
+
+.. note::
+
+    All arguments and return values must be JSON serializable, and round-trip
+    as the same types.
+
+    This means complex types like model instances, as well as many built-in types
+    like ``datetime`` and ``tuple`` cannot be used in tasks without additional
+    conversion.
+
+The ``task`` decorator accepts a few keyword arguments to customize the task:
+
+* ``priority``: The priority of the task. Higher numbers will be executed sooner.
+
+* ``queue_name``: The name of the queue the task will be executed on
+
+* ``backend``: The name of the backend this task must use (as defined in
+  :setting:`TASKS`).
+
+* ``enqueue_on_commit``: Whether the task is enqueued when the current transaction
+  commits successfully, or enqueued immediately.
+
+These arguments correspond to attributes on the created
+:class:`Task <django.tasks.Task>`.
+
+By convention, tasks should be defined in a ``tasks.py`` file, however this is
+not enforced.
+
+Modifying tasks
+---------------
+
+Before enqueueing tasks, it may be necessary to modify certain parameters of the task.
+For example, to give it a higher priority than it would normally.
+
+A ``Task`` instance cannot be modified directly. Instead, a modified instance
+can be created with the :meth:`using <django.tasks.Task.using>` method, leaving
+the original as-is::
+
+    calculate_meaning_of_life.priority  # 0
+
+    calculate_meaning_of_life.using(priority=10).priority  # 10
+
+``using`` allows modifying the following attributes:
+
+* :attr:`priority <django.tasks.Task.priority>`
+* :attr:`backend <django.tasks.Task.backend>`
+* :attr:`queue_name <django.tasks.Task.queue_name>`
+* :attr:`run_after <django.tasks.Task.run_after>`
+
+``run_after`` may also be provided as a :class:`timedelta <datetime.timedelta>`,
+which is used relative to the current time (when ``using`` is called), or a
```

Comment:
Using `timedelta` was a convenience. I agree it could be surprising, and probably wants documenting.

I'll leave this comment here as to whether others think it's a useful API to have, or not worth the footgun.